examples/data-objects/task-selection/src/oldestClientDiceRoller.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { OldestClientObserver } from "@fluid-experimental/task-manager";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IEvent } from "@fluidframework/common-definitions";
+import { assert } from "@fluidframework/common-utils";
+
+import { IDiceRoller } from "./interface";
+
+// The root is map-like, so we'll use this key for storing the value.
+const diceValueKey = "diceValue";
+
+/**
+ * The DiceRoller is our data object that implements the IDiceRoller interface.
+ */
+export class OldestClientDiceRoller extends DataObject implements IDiceRoller {
+    public static get ComponentName() { return "@fluid-example/oldest-client-dice-roller"; }
+
+    private _oldestClientObserver: OldestClientObserver | undefined;
+    private autoRollInterval: ReturnType<typeof setInterval> | undefined;
+
+    /**
+     * initializingFirstTime is run only once by the first client to create the DataObject.  Here we use it to
+     * initialize the state of the DataObject.
+     */
+    protected async initializingFirstTime() {
+        this.root.set(diceValueKey, 1);
+    }
+
+    /**
+     * hasInitialized is run by each client as they load the DataObject.  Here we use it to set up usage of the
+     * DataObject, by registering an event listener for dice rolls.
+     */
+    protected async hasInitialized() {
+        this.root.on("valueChanged", (changed) => {
+            if (changed.key === diceValueKey) {
+                // When we see the dice value change, we'll emit the diceRolled event we specified in our interface.
+                this.emit("diceRolled");
+            }
+        });
+
+        // We can instantiate an OldestClientObserver using an IFluidDataStoreRuntime.
+        this._oldestClientObserver = new OldestClientObserver(this.runtime);
+
+        this.volunteerForAutoRoll();
+    }
+
+    private get oldestClientObserver() {
+        assert(this._oldestClientObserver !== undefined, "OldestClientObserver not initialized");
+        return this._oldestClientObserver;
+    }
+
+    public get value() {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        return this.root.get(diceValueKey);
```

Comment:
cast to number to make code downstream easier?

Reply:
Sounds good.